react_in_rails/react_routing_in_rails.md

```diff
@@ -0,0 +1,212 @@
+- $ bundle add bootstrap
+- Restart the rails server if running!!
+- Modify the *application.css* to be a "sass" file:
+- $ mv app/assets/stylesheets/application.css app/assets/stylesheets/application.scss
+- Add an import to the "sass" file  `@import 'bootstrap'`
+- $ yarn add reactstrap
+
+### Troubleshooting
+- Stop the server and start it again.
+- Did all the setup commands run properly? The commands can be rerun if something isn't working.
+- Seeing a blank page? Look at the terminal for errors.
+- Did you follow all the Rails naming conventions?
+- Did something unexpected happen? Don't be afraid to start over. If you are going to start over make sure to delete the current project completely.
+
+### Step 6: Add Reactstrap Navigation Components
+Using code directly from the Reactstrap docs, add the navigation code to *App.js*
+
+**app/javascript/components/App.js**
+
+```javascript
+import React from "react"
+import PropTypes from "prop-types"
+// add imports for Reactstrap components
+import { Nav, NavItem, NavLink } from 'reactstrap'
+
+class App extends React.Component{
+  render(){
+    return(
+      <React.Fragment>
+        <h1>This is a React Component</h1>
+        <Nav>
+          <NavItem>
+            <NavLink href="#">Link</NavLink>
+          </NavItem>
+          <NavItem>
+            <NavLink disabled href="#">Disabled Link</NavLink>
+          </NavItem>
+        </Nav>
+      </React.Fragment>
+    )
+  }
+}
+
+export default App
+```
+
+### Step 7: Additional React Components
+At this point, there is only one React component in the application. Just like in a regular React App, our project will have many, many components. To keep the components organized we will add two directories: pages and components.
+
+- *App.js* will contain all of our routes
+- *pages* directory will contain the main pages of the app
+- *components* directory will contain all the helper components needed to create beautiful, engaging, user friendly interactions
+
+Due to Rails convention, the React file structure lives in the *app/javascript/components* directory. **Note:** While it may seem a bit confusing, the React components directory will be nested inside of the Rails components directory.
+
+
+### Step 8: Add pages
+
+Now that we have a React file structure and code for navigation we need pages.
+- Add three files in the pages directory: "About Us", "Learn More", and "Home"
+- Add an `<h1>` describing their intent
+
+**app/javascript/components/pages/Home.js**
+
+```javascript
+import React from "react"
+
+class Home extends React.Component {
+  render() {
+    return(
+      <h1>This is the Home Page</h1>
+    )
+  }
+}
+
+export default Home
+```
+
+The other two are mostly the same.
+
+
+### Step 9: React Router
+In order to have multiple pages in a single page app, we need to add the React Router.
+- $ yarn add react-router-dom
+- Import the router to *App.js* `import { BrowserRouter as  Router, Route, Link } from 'react-router-dom'`
```

Comment:
I think you want to use the triple ` here:
` ` `import { BrowserRouter as  Router, Route, Link } from 'react-router-dom'` ` `

```import { BrowserRouter as  Router, Route, Link } from 'react-router-dom'```